Compiler Error on complex nested type

When playing around with complex nested generic types, I found a problem with the following:

```
(deftype (Mop a) [buckets (Array (Array (Pair a a)))])

(defn main []
  (IO.println &(str &(the (Mop Int) (Mop.init [])))))
```

More specifically, this generates invalid C. The compiler errors that I’m getting when trying to compile this are:

```
/Users/veitheller/.carp/out//main.c:293:19: warning: type specifier missing,
      defaults to 'int' [-Wimplicit-int]
typedef Mop__int(*Fn__Array__Array__Pair__int_int_Mop__int)(Array__Array...
                  ^
/Users/veitheller/.carp/out//main.c:293:9: warning: type specifier missing,
      defaults to 'int' [-Wimplicit-int]
typedef Mop__int(*Fn__Array__Array__Pair__int_int_Mop__int)(Array__Array...
~~~~~~~ ^
/Users/veitheller/.carp/out//main.c:293:17: error: function cannot return
      function type 'int (Array__Array__Pair__int_int)' (aka 'int (Array)')
typedef Mop__int(*Fn__Array__Array__Pair__int_int_Mop__int)(Array__Array...
                ^
/Users/veitheller/.carp/out//main.c:630:23: error: function cannot return
      function type 'Mop__int' (aka 'int (int *)')
Mop__int Mop_init__int(Array__Array__Pair__int_int buckets);
                      ^
/Users/veitheller/.carp/out//main.c:1836:41: error: member reference base type
      'Mop__int *' (aka 'int (*)(int *)') is not a structure or union
    Array_delete__Array__Pair__int_int(p.buckets);
                                       ~^~~~~~~~
/Users/veitheller/.carp/out//main.c:1839:23: error: function cannot return
      function type 'Mop__int' (aka 'int (int *)')
Mop__int Mop_init__int(Array__Array__Pair__int_int buckets) {
                      ^
/Users/veitheller/.carp/out//main.c:1841:13: error: member reference base type
      'Mop__int' (aka 'int (int *)') is not a structure or union
    instance.buckets = buckets;
    ~~~~~~~~^~~~~~~~
/Users/veitheller/.carp/out//main.c:1842:12: warning: incompatible pointer to
      integer conversion returning 'Mop__int' (aka 'int (int *)') from a
      function with result type 'int' [-Wint-conversion]
    return instance;
           ^~~~~~~~
/Users/veitheller/.carp/out//main.c:1851:44: error: member reference base type
      'Mop__int' (aka 'int (int *)') is not a structure or union
  temp = Array_prn__Array__Pair__int_int(&p->buckets); 
                                          ~^ ~~~~~~~
/Users/veitheller/.carp/out//main.c:1862:44: error: member reference base type
      'Mop__int' (aka 'int (int *)') is not a structure or union
  temp = Array_prn__Array__Pair__int_int(&p->buckets);
                                          ~^ ~~~~~~~
/Users/veitheller/.carp/out//main.c:2570:14: error: illegal initializer (only
      variables can be initialized)
    Mop__int _14 = _13; // From the 'the' function.
             ^
3 warnings and 8 errors generated.
```

Apparently the ordering of `typedef`s  is off.

Cheers

--- a/src/Scoring.hs
+++ b/src/Scoring.hs
@@ -15,7 +15,10 @@ scoreTypeBinder typeEnv b@(Binder _ (XObj (Lst (XObj x _ _ : XObj (Sym _ _) _ _ 
   case x of
     Defalias aliasedType ->
       let selfName = ""
-      in  (depthOfType typeEnv selfName aliasedType, b)
+      -- we add 1 here because deftypes generate aliases that
+      -- will at least have the same score as the type, but
+      -- need to come after. the increment represents this dependency
+      in  (depthOfType typeEnv selfName aliasedType + 1, b)
     Typ (StructTy structName varTys) ->
       case lookupInEnv (SymPath [] structName) (getTypeEnv typeEnv) of
         Just (_, Binder _ typedef) -> let depth = ((depthOfDeftype typeEnv typedef varTys), b)

